eos-phone-home

```diff
@@ -0,0 +1,398 @@
+
+        command_line = None
+        with open('/proc/cmdline', 'r') as command_line_file:
+            command_line = command_line_file.read()
+
+        if command_line:
+            command_line = command_line.strip()
+
+        print(" - Cmdline:", command_line)
+
+        if re.search(r'\bendless.live_boot\b', command_line):
+            print(" - Found live marker. Marking as live image.")
+            live = True
+
+        return live
+
+    def _get_count(self):
+        count = None
+
+        try:
+            with open(os.path.join(self.STATE_DIRECTORY, 'count'), 'r') \
+              as count_file:
+                count_data = count_file.read()
+
+                if count_data:
+                    count = int(count_data.strip())
+
+            print(" - Count:", count)
+        except Exception as e:
+            print("Unable to get count!", e, file=sys.stderr)
+
+        return count
+
+    def _set_count(self, count):
+        try:
+            with open(os.path.join(self.STATE_DIRECTORY, 'count'), 'w') \
+              as count_file:
+                count_file.write("%d" % count)
+
+            self._variables['count'] = count
+
+            print(" - Updated count:", count)
+        except Exception as e:
+            print("Unable to set count!", e, file=sys.stderr)
+
+    def _get_var(self, name):
+        if name not in self._variables:
+            print("Getting variable:", name)
+            var = getattr(self, '_get_' + name)()
+
+            if var is None or var == '':
+                var = "unknown"
+
+            self._variables[name] = var
+
+        return self._variables[name]
+
+    def _send_to_server(self, endpoint, variables):
+        request = {}
+        for name in variables:
+            request[name] = self._get_var(name)
+
+        request_data = json.dumps(request)
+        print("Request data:", request_data)
+
+        print("Sending to server (%s)..." % endpoint)
+
+        if self._debug:
+            print("Debugging turned on so nothing sent to server!")
+            return True
+
+        success = False
+
+        try:
+            data = request_data.encode('utf-8')
+
+            req = urllib.request.Request(endpoint, data=data, method='PUT')
+            req.add_header('Content-Type', 'application/json')
+
+            response_data = None
+            with urllib.request.urlopen(req) as conn:
+                response_data = conn.read().decode('utf-8')
+
+            print('Response:', response_data)
+
+            response = json.loads(response_data)
+
+            if 'success' in response and \
+               response['success'] == True:
+                print("Server returned success message!")
+                success = True
+            else:
+                print("Server failed to process the data:", response)
+
+        except Exception as e:
+            print("Sending failed!", e, file=sys.stderr)
+
+        return success
+
+    def _need_activate(self):
```

Comment:
`_is_activated` or `_needs_activation` - the current name seems grammatically poor.